versions/schematic-2.md

```diff
@@ -0,0 +1,235 @@
+                Amount: -0.05492870821106577d,
+                Operation: 1,
+                Name: "Random spawn bonus"
+            }
+        ],
+        Name: "generic.followRange" }
+    ],
+    Invulnerable: 0b,
+    FallFlying: 0b,
+    ForcedAge: 0,
+    PortalCooldown: 0,
+    AbsorptionAmount: 0.0f,
+    FallDistance: 0.0f,
+    InLove: 0,
+    DeathTime: 0s,
+    HandDropChances: [ 0.085f, 0.085f ],
+    PersistenceRequired: 1b,
+    Spigot.ticksLived: 74919199,
+    Age: 0,
+    Motion: [ 0.0d, -0.0784000015258789d, 0.0d ],
+    Leashed: 0b,
+    UUIDLeast: -7755591127829794293L,
+    Health: 10.0f,
+    LeftHanded: 0b,
+    Air: 300s,
+    OnGround: 1b,
+    Dimension: 0,
+    Rotation: [ 194.99161f, 0.0f ],
+    HandItems: [ {}, {} ],
+    ArmorDropChances: [ 0.085f, 0.085f, 0.085f, 0.085f ],
+    UUIDMost: 7781923390897078472L,
+    Pos: [ 121.25787019796437d, 69.0d, 139.3946334861942d ],
+    Fire: -1s,
+    ArmorItems: [ {}, {}, {}, {} ],
+    CanPickUpLoot: 0b,
+    HurtTime: 0s
+}
+```
+
+## Changelog
+
+Following changes has been made compared to version 1:
+
+### Functional changes
+
+* Changed [Palette](#schematicPalette) from `object` with IDs as key and index as value to a `list` with IDs, so you can easily calculate an index and **access it directly without searching**.
+* Changed [BlockData](#schematicBlockData) from `varint[]` to `int[]` or `long[]` to support even bigger areas and because **NBT has no varint or varlong definition**.
+* Changed [Width](#schematicWidth), [Height](#schematicHeight), [Length](#schematicLength) from `unsigned short` to `unsigned integer` to support even **bigger areas and noone cares about 3 more bytes**.
+* Changed [PaletteMax](#schematicPaletteMax) from `integer` to `byte`, because **8 would be the maximum** possible value for Java implementations (64-bit long)
+* Added [Entity](#entityObject) field to **support entities**.
+* Changed [Date](#metadataDate) from miliseconds to seconds to the default **UNIX timestamp**, to cover a longer time range and because noone needs miliseconds.
+* Moved [ContentVersion](#schematicContentVersion) from entity & tile compounds up to the root compound. Because there is only one relevant version when exported, there is **no need for versioning each tile & entity**.
+* Changed [Pos](#tileEntityPosX) of tiles to x, y, z. Because mojangs specified it like this, so there is **no more conversion necessary**.
+* Added [OptionalMods](#metadataOptionalMods) to Metadata. Because there often are only optional additional **features within tiles or entities, which are not necessarily needed**.
+
+### Other changes
+
+* Added **OPTIONAL** or **REQUIRED** in every field to make it more clear
+* Replaced the word `region` by the word `area`, to avoid misunderstanding since a minecraft world already has so called `regions`
+* Made the offset more clear
+* Moved Definition#BlockState to chapter PaletteObject
+* Added the corresponding NBT type to every field
```

Comment:
Sure, you can represent each field as an `NBT type`, but the format can already define the number formats used throughout the document and explain *in a different table* of what that means.

Case in point:

Name | Type | NBT Equivalent
---|:---:|---
<a name="short"></a>Short | `short` | Stored as NBT Short
<a name="unsigned short"></a>Unsigned Short | `unsigned short` | Stored as NBT Short and absolute valued (values being from 0 -> 2^16 -1)
<a name="varint"></a>Varint | `varint` | Stored as NBT byte with the integer value being packed and the length being the first byte of data.

Reply:
I like the idea of a format specification table to define the formats used within the document.
But i don't understand how you would store a varint as NBT byte.
Are we talking about https://wiki.vg/Protocol#VarInt_and_VarLong?
Do you mean by "packed" append all following bytes? But then it would be a NBT byte array, wouldn't it?